core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java

```diff
@@ -293,4 +309,59 @@ public void testMergeAppend() throws Exception {
     Assert.assertEquals("Current snapshot should contain 1 merged manifest",
         1, metadata.currentSnapshot().manifests().size());
   }
+
+  @Test
+  public void testRenameFailures() throws Exception {
+    Assert.assertTrue("Should create v1 metadata",
+        version(1).exists() && version(1).isFile());
+    Assert.assertFalse("Should not create v2 or newer versions",
+        version(2).exists());
+    assertTrue(table instanceof BaseTable);
+    final BaseTable bt = (BaseTable) table;
```

Comment:
Please remove unnecessary `final` modifiers.

Reply:
It would be better to use a longer variable name, like `base`. `b` doesn't clearly mean "base" later on.